### PR TITLE
Added setFormValues function to validaiton plugin

### PR DESCRIPTION
**Pull request by priyankjalu89**
- [x] Does each function have a comment to describe what it does ?
- [x] Are there comments throughout the code ?
- [x] Is each function unit testable ?
- [x] Are the function or method names well chosen ?
- [x] Are variable names well chosen ?

### CODE REVIEW

File: js/jquery.validation.js [hunk 1]
```diff
@@ -438,6 +438,20 @@
...
             }
         }
+
+	setValuesFromResponse: function(response) {
+	    // Set form values from the response
+	    var $form = $(this.element);
+	    if (typeof response !== "undefined" && 'record' in response ) {
+		$.each(response.record, function (name, object) {
+		    var $element = $form.find('[name=' + name + ']');
+		    if ( 'valid' in object && object.valid && 'value' in object ) {
```
Comment (daniel-qcode) [inline js/jquery.validation.js]: In your codepen can you add the following types of form elements :
 * text input
 * select dropdown
 * checkbox with value = true
 * checkbox with value = "value1"

Then do some testing to check that this functions correctly sets the value of the different types of input above.

I suspect that .val() might not update the checkboxes as expected but could be wrong.

**Comment by daniel-qcode** [comment]
Think it might be fine as is ... the current behaviour of dbForm doesn't take responsibility for checking checkboxes based on the response it just sets the value too.

Does text and select dropdown work as expected ?

**Comment by priyankjalu89** [comment]
Yes text and select dropdown works as expected.

**Comment by bvw** [comment]
### Suggested behaviour

value=true && json=true then tick checkbox
value=true && json=false then untick tickbox

value=foo && json=foo then tick checkbox
value=foo && json<>foo then tick uncheckbox

#### Casting to boolean

value=yes && json=true then tick checkbox
value=yes && json=false then untick tickbox

value=t && json=yes then tick checkbox
value=t && json=no then untick tickbox

castBoolean(value)=true && castBoolean(json)=true then tick checkbox
castBoolean(value)=true && castBoolean(json)=false then untick tickbox

**Comment by daniel-qcode** [comment]
Could we use existing qcode-ui `parseBoolean` function to help with this:
https://github.com/qcode-software/qcode-ui/blob/fdd9bae525a80379ce4d42800789c603cf561bd9/js/jquery.utils.js#L211

**Comment by daniel-qcode** [comment]
@bvw So just to confirm the suggested behaviour for checkboxes is to check the checkboxes if the values match (with or without casting to boolean) but we should not be setting the value of the checkbox itself.
I think that makes sense,  is that consistent with what you are proposing. ?

**Pull request merged**